src/backend/distributed/commands/call.c

```diff
@@ -217,3 +202,83 @@ CallFuncExprRemotely(CallStmt *callStmt, DistObjectCacheEntry *procedure,
 
 	return true;
 }
+
+
+/*
+ * ShardPlacementWhenColocatedWithReferenceTable decides on a placement for delegating
+ * a procedure call that reads from a reference table.
+ *
+ * If citus.task_assignment_policy is set to round-robin, we assign a different placement
+ * on consecutive runs. Otherwise the function returns the first placement available.
+ */
+static ShardPlacement *
+ShardPlacementWhenColocatedWithReferenceTable(CitusTableCacheEntry *cacheEntry)
+{
+	const ShardInterval *shardInterval = cacheEntry->sortedShardIntervalArray[0];
+	const uint64 referenceTableShardId = shardInterval->shardId;
+	List *placementList = ActiveShardPlacementList(referenceTableShardId);
+
+	if (TaskAssignmentPolicy == TASK_ASSIGNMENT_ROUND_ROBIN)
+	{
+		placementList = RoundRobinReorder(NULL, placementList);
+	}
+
+	return (ShardPlacement *) linitial(placementList);
+}
+
+
+/*
+ * ShardPlacementWhenColocatedWithDistTable decides on a placement
+ * for delegating a procedure call that accesses a distributed table.
+ */
+static ShardPlacement *
+ShardPlacementWhenColocatedWithDistTable(DistObjectCacheEntry *procedure,
+										 FuncExpr *funcExpr,
+										 Var *partitionColumn,
+										 CitusTableCacheEntry *cacheEntry)
+{
+	if (procedure->distributionArgIndex < 0 ||
+		procedure->distributionArgIndex >= list_length(funcExpr->args))
+	{
+		ereport(DEBUG1, (errmsg("cannot push down invalid distribution_argument_index")));
```

Comment:
Shouldn't we do as below, it seems more like a bug catch ?

```suggestion
		ereport(ERROR, (errcode(ERRCODE_INTERNAL_ERROR),
						           errmsg("cannot push down invalid distribution_argument_index")));
```

Reply:
We still can execute the procedure call on the coordinator worker, if we can not delegate it on a worker. I think it's safer to have a debug message and return NULL, and the caller will execute the CALL on the coordinator